test/e2e/network/networking_perf.go

```diff
@@ -160,3 +172,227 @@ var _ = SIGDescribe("Networking IPerf [Experimental] [Slow] [Feature:Networking-
+		framework.ExpectNoError(err, "deploy iperf2 client daemonset")
+
+		clientPodsListOptions := metav1.ListOptions{
+			LabelSelector: "app=iperf2-client",
+		}
+
+		framework.ExpectNoError(e2edeployment.WaitForDeploymentComplete(f.ClientSet, serverDeployment), "unable to wait for server deployment")
+
+		err = wait.PollImmediate(2*time.Second, 100*time.Second, func() (done bool, err error) {
+			clientPodList, err := f.ClientSet.CoreV1().Pods(f.Namespace.Name).List(context.TODO(), clientPodsListOptions)
+			if err != nil {
+				return false, err
+			}
+			if len(clientPodList.Items) < len(nodeList.Items) {
+				return false, nil
+			}
+			for _, pod := range clientPodList.Items {
+				if pod.Status.Phase != v1.PodRunning {
+					return false, nil
+				}
+			}
+			return true, nil
+		})
+		framework.ExpectNoError(err, "unable to wait for client pods to come up")
+
+		serverPodList, err := f.ClientSet.CoreV1().Pods(f.Namespace.Name).List(context.TODO(), serverPodsListOptions)
+		framework.ExpectNoError(err)
+		if len(serverPodList.Items) != 1 {
+			framework.Failf("expected 1 server pod, found %d", len(serverPodList.Items))
+		}
+		serverPod := serverPodList.Items[0]
+
+		clientPodList, err := f.ClientSet.CoreV1().Pods(f.Namespace.Name).List(context.TODO(), clientPodsListOptions)
+		framework.ExpectNoError(err)
+		framework.Logf("found %d matching client pods", len(clientPodList.Items))
+		if len(clientPodList.Items) == 0 {
+			framework.Failf("found 0 matching client pods")
+		}
+
+		nodeResults := &iperf2Results{
+			ServerNode: serverPod.Spec.NodeName,
+			Results:    map[string]*iperf2NodeResults{},
+		}
+		for _, pod := range clientPodList.Items {
+			podName := pod.Name
+			hostIP := pod.Status.HostIP
+
+			command := fmt.Sprintf(`iperf -e -p 6789 --reportstyle C -i 1 -c iperf2-server -T "Client on %s"`, hostIP)
+			framework.Logf("attempting to run command '%s' in pod %s", command, podName)
+			output := f.ExecShellInPod(podName, command)
+			framework.Logf("output from exec on pod %s: \n%s\n", podName, output)
+
+			var parsedResults []*IPerfResult
+			for _, line := range strings.Split(output, "\n") {
+				parsedResults = append(parsedResults, NewIPerf(line))
+			}
+			if parsedResults == nil || len(parsedResults) == 0 {
+				framework.Failf("no results parsed from iperf output from client %s", pod.Name)
+			}
+			results := &iperf2NodeResults{
+				Intervals: parsedResults[:len(parsedResults)-1],
+				Total:     parsedResults[len(parsedResults)-1],
+			}
+			nodeResults.Results[pod.Spec.NodeName] = results
+		}
+		framework.Logf("%25s%25s%25s", "From", "To", "Bandwidth (MB/s)")
+		for nodeFrom, results := range nodeResults.Results {
+			mb := 1024 * 1024
+			bandwidthMB := math.Round(float64(results.Total.bandwidthBits) / float64(mb))
+			framework.Logf("%25s%25s%25.0f", nodeFrom, nodeResults.ServerNode, bandwidthMB)
+		}
+		for nodeFrom, results := range nodeResults.Results {
+			if results.Total.bandwidthBits < eightBillion {
```

Comment:
After changing the const name on the bottom, using the same const (with the unit in the name) here will make things more clear

Reply:
👍